core/src/main/java/org/apache/iceberg/rest/auth/OAuth2Manager.java

```diff
@@ -259,16 +260,41 @@ protected OAuth2Util.AuthSession newSessionFromTokenExchange(
 
   protected OAuth2Util.AuthSession newSessionFromTokenResponse(
       AuthConfig config, OAuth2Util.AuthSession parent) {
+    RESTClient client = refreshClient;
     OAuthTokenResponse response =
         OAuth2Util.fetchToken(
-            refreshClient,
+            client,
             Map.of(),
             config.credential(),
             config.scope(),
             config.oauth2ServerUri(),
             config.optionalOAuthParams());
     return OAuth2Util.AuthSession.fromTokenResponse(
-        refreshClient, refreshExecutor(), response, System.currentTimeMillis(), parent);
+        client, refreshExecutor(), response, System.currentTimeMillis(), parent);
+  }
+
+  private void maybeInitRefreshClient(RESTClient sharedClient, AuthSession session) {
+    if (refreshClient == null) {
+      synchronized (this) {
+        if (refreshClient == null) {
+          refreshClient = sharedClient.withAuthSession(session);
+        }
+      }
+    }
+  }
+
+  private AuthSessionCache getOrCreateSessionCache(Map<String, String> properties) {
+    AuthSessionCache cache = this.sessionCache;
```

Comment:
I think this can be simplified to
```
private AuthSessionCache sessionCache(Map<String, String> properties) {
    if (sessionCache == null) {
      synchronized (this) {
        if (sessionCache == null) {
          this.sessionCache = newSessionCache(name, properties);
        }
      }
    }

    return sessionCache;
  }
```